src/file_pipe_log/mod.rs

```diff
@@ -20,60 +21,154 @@ pub mod debug {
     //! A set of public utilities used for interacting with log files.
 
     use std::collections::VecDeque;
+    use std::io::{Read, Result as IoResult, Seek, SeekFrom, Write};
     use std::path::{Path, PathBuf};
     use std::sync::Arc;
 
-    use crate::file_builder::FileBuilder;
+    use crate::file_pipe_log::log_file::LogFd;
+    use crate::file_system::{FileSystem, LowExt, ReadExt, WriteExt};
     use crate::log_batch::LogItem;
     use crate::pipe_log::FileId;
     use crate::{Error, Result};
 
     use super::format::FileNameExt;
-    use super::log_file::{LogFd, LogFileReader, LogFileWriter};
+    use super::log_file::{LogFileReader, LogFileWriter};
     use super::reader::LogItemBatchFileReader;
 
+    pub struct ObfuscatedFile {
```

Comment:
I think you can implement `ObfuscatedFileSystem` outside of `file_pipe_log` (inside "file_system.rs" maybe), i.e. avoid using low-level details like `LogFd`.

Reply:
there is my plan:  
- make a `env` directory.  
-  move `FileSystem`, `DefaultFileSystem`,`LogFd`, `LogFdFile`, `ObfuscatedFileSystem ` into it. 
- `LogFileWriter`, `LogFileReader` rename to `LogWriter` and `LogReader` keep still in `file_pipe_log` directory 
- `file_pipe_log` depends  on `env`.`FileSystem` abstraction. 
- but `ObfuscatedFileSystem` still use `LogFd` as its `Handler`

any suggestion about it?

Reply:
LGTM. But I still want `ObfuscatedFileSystem` to be stacked over `DefaultFileSystem`. This way, you don't need to keep track of `offset` in `ObfuscatedFile`, but instead simply use `struct  ObfuscatedReader(DefaultFileSystem::Reader)`.

Reply:
Done